c10/core/SingletonSymNodeImpl.h

```diff
@@ -0,0 +1,121 @@
+#include <c10/core/ConstantSymNodeImpl.h>
+#include <c10/core/SymBool.h>
+#include <c10/core/SymNodeImpl.h>
+#include <iostream>
+
+namespace c10 {
+
+// An object that only defines the equality operator.
+class C10_API SingletonSymNodeImpl : public SymNodeImpl {
+ public:
+  SingletonSymNodeImpl(int64_t val) : val_(val) {}
```

Comment:
Gotta be very careful with this constructor, people might think this denotes regular int semantics

Reply:
Also related https://github.com/pytorch/pytorch/pull/107089#discussion_r1294006232? 
Would it be sufficient to have a comment saying that no one should call this constructor directly?

Reply:
Made the construct explicit and added a comment telling folks to not use this directly.